pkg/informers/service/informer_service.go

```diff
@@ -145,3 +147,33 @@ func (s *ServiceImpl) GetNSTemplateTier(name string) (*toolchainv1alpha1.NSTempl
 	}
 	return tier, err
 }
+
+func (s *ServiceImpl) BannedUsersByEmail(email string) ([]toolchainv1alpha1.BannedUser, error) {
+	hashedEmail := hash.EncodeString(email)
+	r, err := labels.NewRequirement(toolchainv1alpha1.BannedUserEmailHashLabelKey, selection.Equals, []string{hashedEmail})
+	if err != nil {
+		return nil, err
+	}
+
+	ls := labels.NewSelector().Add(*r)
+	objs, err := s.informer.BannedUsers.ByNamespace(configuration.Namespace()).List(ls)
+	if err != nil {
+		return nil, err
+	}
+
+	bb := []toolchainv1alpha1.BannedUser{}
+	for _, obj := range objs {
+		unobj := obj.(*unstructured.Unstructured)
+		bu := &toolchainv1alpha1.BannedUser{}
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unobj.UnstructuredContent(), bu); err != nil {
+			log.Errorf(nil, err, "failed to list BannedUsers")
+			return nil, err
+		}
+
+		// check in case of hash collision
+		if bu.Spec.Email == email {
+			bb = append(bb, *bu)
+		}
+	}
+	return bb, nil
+}
```

Comment:
Can you please add tests for this new function to `informer_service_test.go`?

Reply:
added in e2625007b558f2d2fa131df46e0f9ffa5ef238a0